utils/utils.go

```diff
@@ -125,4 +128,188 @@ func HandlePanic() {
+		}
+	} else if osName == "linux" {
+		*keployAlias = "sudo docker run --pull always --name keploy-v2 -e BINARY_TO_DOCKER=true -p 16789:16789 --privileged --pid=host -it -v " + os.Getenv("PWD") + ":" + os.Getenv("PWD") + " -w " + os.Getenv("PWD") + " -v /sys/fs/cgroup:/sys/fs/cgroup -v /sys/kernel/debug:/sys/kernel/debug -v /sys/fs/bpf:/sys/fs/bpf -v /var/run/docker.sock:/var/run/docker.sock -v " + os.Getenv("HOME") + "/.keploy-config:/root/.keploy-config -v " + os.Getenv("HOME") + "/.keploy:/root/.keploy --rm ghcr.io/keploy/keploy "
+	}
+}
+
+func appendFlags(flagName string, flagValue string) string {
+	if len(flagValue) > 0 {
+		// Check for = in the flagName.
+		if strings.Contains(flagName, "=") {
+			return " --" + flagName + flagValue
+		}
+		return " --" + flagName + " " + flagValue
+	}
+	return ""
+}
+
+func UpdateKeployToDocker(cmdName string, isDockerCompose bool, flags interface{}, logger *zap.Logger) {
+	var recordFlags RecordFlags
+	var testFlags TestFlags
+	//Check the type of flags.
+	switch flag := flags.(type) {
+	case RecordFlags:
+		recordFlags = flag
+	case TestFlags:
+		testFlags = flag
+	default:
+		logger.Error("Unknown flags provided")
+		return
+	}
+	var keployAlias string
+	getAlias(&keployAlias, logger)
+	keployAlias = keployAlias + cmdName + " -c "
+	var cmd *exec.Cmd
+	if cmdName == "record" {
+		keployAlias = keployAlias + "\"" + recordFlags.Command + "\" "
+		if len(recordFlags.PassThroughPorts) > 0 {
+			portSlice := make([]string, len(recordFlags.PassThroughPorts))
+			for i, port := range recordFlags.PassThroughPorts {
+				portSlice[i] = fmt.Sprintf("%d", port)
+			}
+			joinedPorts := strings.Join(portSlice, ",")
+			keployAlias = keployAlias + " --passThroughPorts=" + fmt.Sprintf("%v ", joinedPorts)
+		}
+		if recordFlags.ConfigPath != "." {
+			keployAlias = keployAlias + " --configPath " + recordFlags.ConfigPath
+		}
+		if len(recordFlags.Path) > 0 {
+			keployAlias = keployAlias + " --path " + recordFlags.Path
+		}
+		addtionalFlags := appendFlags("containerName", recordFlags.ContainerName) + appendFlags("buildDelay ", recordFlags.BuildDelay.String()) + appendFlags("delay", fmt.Sprintf("%d", recordFlags.Delay)) + appendFlags("proxyport", fmt.Sprintf("%d", recordFlags.Proxyport)) + appendFlags("networkName", recordFlags.NetworkName) + appendFlags("enableTele=", fmt.Sprintf("%v", recordFlags.EnableTele))
+		keployAlias = keployAlias + addtionalFlags
+		cmd = exec.Command("sh", "-c", keployAlias)
+
+	} else {
+		keployAlias = keployAlias + "\"" + testFlags.Command + "\" "
+		if len(testFlags.PassThroughPorts) > 0 {
+			portSlice := make([]string, len(recordFlags.PassThroughPorts))
+			for i, port := range recordFlags.PassThroughPorts {
```

Comment:
It should be `testFlags` rather than `recordFlags` while setting `passThroughPorts` flag.

Reply:
done